src/transformers/models/patchtsmixer/modeling_patchtsmixer.py

```diff
@@ -0,0 +1,2362 @@
+                time series.
+
+                For univariate time series, `input_size` dimension should be 1. For multivariate time series, it is
+                greater than 1.
+            observed_mask (`torch.FloatTensor` of shape `(batch_size, sequence_length, input_size)`, *optional*):
+                Boolean mask to indicate which `past_values` were observed and which were missing. Mask values selected
+                in `[0, 1]`:
+                    - 1 for values that are **observed**,
+                    - 0 for values that are **missing** (i.e. NaNs that were replaced by zeros).
+
+            output_hidden_states (`bool`, *optional*):
+                Whether or not to return the hidden states of all layers.
+            return_dict (`bool`, *optional*):
+                Return dict.
+
+        Returns:
+            [`PatchTSMixerModelOutput`]
+        """
+
+        mask = None
+        if observed_mask is None:
+            observed_mask = torch.ones_like(context_values)
+        scaled_context_values, loc, scale = self.scaler(context_values, observed_mask)
+
+        patched_x = self.patching(scaled_context_values)  # [batch_size x input_size x num_patch x patch_len
+
+        enc_input = patched_x
+        if self.masking is not None:
+            enc_input, mask = self.masking(patched_x)
+            # enc_input: [batch_size x input_size x num_patch x patch_len]
+            # mask: [batch_size x input_size x num_patch]
+
+        encoder_output = self.encoder(enc_input, output_hidden_states=output_hidden_states)
+
+        return PatchTSMixerModelOutput(
+            last_hidden_state=encoder_output.last_hidden_state,
+            hidden_states=encoder_output.hidden_states,
+            patched_input=patched_x,
+            mask=mask,
+            loc=loc,
+            scale=scale,
+        )
+
+
+@dataclass
+class PatchTSMixerForMaskPreTrainingOutput(ModelOutput):
+    """
+    Output type of [`PatchTSMixerForMaskPreTrainingOutput`].
+
+    Args:
+        prediction_logits (`torch.FloatTensor` of shape `(batch_size, input_size, num_patches, patch_len)`):
+            Prediction output from the pretrain head.
+        hidden_states (`tuple(torch.FloatTensor)`, *optional*):
+            Hidden-states of the model at the output of each layer.
+        last_hidden_state (`torch.FloatTensor` of shape `(batch_size, input_size, num_patches, num_features)`):
+            Backbone embeddings before passing through the head.
+        loss (*optional*, returned when `y` is provided, `torch.FloatTensor` of shape `()`):
+            Total loss
+    """
+
+    prediction_logits: torch.FloatTensor = None
+    last_hidden_state: torch.FloatTensor = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    loss: Optional[torch.FloatTensor] = None
+
+
+class PatchTSMixerForMaskPretraining(PatchTSMixerPreTrainedModel):
```

Comment:
see other PR, let's use the same term as BertForPretraining
```suggestion
class PatchTSMixerForPretraining(PatchTSMixerPreTrainedModel):
```